R/zchunk_L1233.Elec_water.R

```diff
@@ -39,89 +38,301 @@ module_water_L1233.Elec_water_DISABLED <- function(command, ...) {
 
-    # Produce outputs
-    # Temporary code below sends back empty data frames marked "don't test"
-    # Note that all precursor names (in `add_precursor`) must be in this chunk's inputs
-    # There's also a `same_precursors_as(x)` you can use
-    # If no precursors (very rare) don't call `add_precursor` at all
-    tibble() %>%
-      add_title("descriptive title of data") %>%
-      add_units("units") %>%
-      add_comments("comments describing how data generated") %>%
-      add_comments("can be multiple lines") %>%
+    # STEP 1. DOWNSCALE WATER COOLING SYSTEM SHARES FROM 14 REGIONS to 201 COUNTRIES
+
+    # RENAME INTERMEDIATE FUELS AND AGGREGATE BY UPDATED FUEL TYPE
+    L101.en_bal_EJ_ctry_Si_Fi_Yh_full %>%
+      filter(sector == "out_electricity generation") %>%
+      left_join(select(enduse_fuel_aggregation, fuel, electricity), by = "fuel") %>%
+      select(-fuel) %>%
+      rename(fuel = electricity) -> L1233.out_EJ_ctry_elec_Fi_Yh_full
+
+    L1233.out_EJ_ctry_elec_Fi_Yh_full %>%
+      filter(year %in% HISTORICAL_YEARS) %>%
+      group_by(iso, year, fuel) %>%
+      summarise(value = sum(value)) %>%
+      ungroup() %>%
+      filter(is.na(fuel) == F) %>%
+      mutate(sector = "electricity generation") -> L1233.out_EJ_ctry_elec_F_Yh
+
+    # INTERPOLATE A23.CoolingSystemShares_RG3 FOR HISTORICAL YEARS AND HISTORICAL + FUTURE YEARS
+    A23.CoolingSystemShares_RG3 %>%
+      gather(year, value, -region_GCAM3, -plant_type, -cooling_system, -water_type) %>%
+      mutate(year = as.integer(year)) -> A23.CoolingSystemShares_RG3_LF
+
+    INTERPOLATED_YEARS <- HISTORICAL_YEARS[HISTORICAL_YEARS %in%
+                                              unique(A23.CoolingSystemShares_RG3_LF$year) == F]
+    EXTRAPOLATED_YEARS <- c(HISTORICAL_YEARS, FUTURE_YEARS)[c(HISTORICAL_YEARS, FUTURE_YEARS) %in%
+                                                              unique(A23.CoolingSystemShares_RG3_LF$year) == F]
+
+    setNames(data.frame(matrix(ncol = length(INTERPOLATED_YEARS),
+                               nrow = nrow(A23.CoolingSystemShares_RG3))),
+             as.character(INTERPOLATED_YEARS)) %>%
+      as_tibble() %>%
+      bind_cols(A23.CoolingSystemShares_RG3) %>%
+      gather(year, value, -region_GCAM3, -plant_type, -cooling_system, -water_type) %>%
+      mutate(year = as.integer(year)) %>%
+      arrange(year) %>%
+      group_by(region_GCAM3, plant_type, cooling_system, water_type) %>%
+      mutate(value = approx_fun(year, value, rule = 1)) %>%
+      ungroup() -> A23.CoolingSystemShares_RG3_LF_interp
+
+    setNames(data.frame(matrix(ncol = length(EXTRAPOLATED_YEARS),
+                               nrow = nrow(A23.CoolingSystemShares_RG3))),
+             as.character(EXTRAPOLATED_YEARS)) %>%
+      as_tibble() %>%
+      bind_cols(A23.CoolingSystemShares_RG3) %>%
+      gather(year, value, -region_GCAM3, -plant_type, -cooling_system, -water_type) %>%
+      mutate(year = as.integer(year)) %>%
```

Comment:
You keep doing this. Why not convert A23.CoolingSystemShares_RG3's `year` column to integer once, when it's read in?